Ajax login

milafrerichs opened the pull request:
## What does this PR do?
Adds the ability to allow login via Ajax once the map is beiing saved.
Needs CORS to work. 
Adds a new option, which is not enabled. `options.checkStatus = true`

## Screenshot
<img width="942" alt="bildschirmfoto 2017-01-30 um 18 47 17" src="https://cloud.githubusercontent.com/assets/688980/22434483/8de1fc16-e71c-11e6-8deb-07c1cde22e1b.png">

## Related Issue
closes #102 
## ToDO
- [x] switch to SDK LoginModal

milafrerichs committed:
Add login method

milafrerichs committed:
FIx reducer tests for mapConfig

milafrerichs committed:
Add userLoggedIn to map state

add action, for setting and retrieving the user and add integration
tests to test the redux path

milafrerichs committed:
Set user logged in if geonode user is logged in

milafrerichs committed:
Add checkLogin

is false by default. only if checkLogin is true the ajax login will be
shown if the user is not logged in.
this needs CORS that's why we need to check if it needs to be checked
and shown

milafrerichs committed:
Add LoginView to saveContainer

only show if the user is not logged in and checkLogin is true
add dispatch action for login. if it succeeds, set is loggedIn to true

milafrerichs committed:
Fix lintin error

milafrerichs committed:
Add methid for close

add more tests, integrate with composer

milafrerichs committed:
Change type of request for login

fix a few flaws with the ajax login

bartvde commented on src/components/loginView.jsx:
```diff
@@ -0,0 +1,109 @@
+    };
+  }
+  getChildContext() {
+    return {muiTheme: getMuiTheme()};
```
this will make the component not listen to custom themes, since it will always use the default theme

milafrerichs replied:
right, custom themes... thanks, need to update it...

bartvde commented on src/components/loginView.jsx:
```diff
@@ -0,0 +1,109 @@
+  },
+});
+
+export class LoginView extends React.Component {
```
was there a way where we could use re-use some of the SDK login modal code, or was this not an option?

milafrerichs committed:
Merge branch 'master' into 112-ajax-login

bartvde commented:
Will work on the change in SDK now to make the login modal more flexible

bartvde commented:
See: https://github.com/boundlessgeo/sdk/pull/368

milafrerichs committed:
Updarte SDK use LoginModal from SDK

instead of defining my own loginModal use the one from the SDk
and update the SDK as well to use the new LoginModal methods

milafrerichs committed:
Merge branch 'master' into 112-ajax-login

bartvde commented on src/components/saveContainer.jsx:
```diff
@@ -1,11 +1,12 @@
 import React from 'react';
-import {injectIntl, intlShape} from 'react-intl';
+import {addLocaleData, injectIntl, intlShape} from 'react-intl';
```
is this used here?

milafrerichs replied:
no, good catch... forgot to delete it, after I tested the change in modal text

milafrerichs committed:
remove localedata import
